app/models/policy/single_bid_auction.rb

```diff
@@ -0,0 +1,110 @@
+class Policy::SingleBidAuction < Policy::Auction
+  TYPE_CODE = 'single_bid'.freeze
+
+  # disallow bids if the user has already placed a bid in addition to
+  # general bidding rules
+  def user_can_bid?
+    super && !user_is_bidder?
+  end
+
+  def bids?
+    return user_is_bidder? if available?
+    super
+  end
+
+  def highlighted_bid
+    if available?
+      user_bids.first
+    else
+      winning_bid
+    end
+  end
+
+  def winning_bid
+    return Presenter::Bid::Null.new if available?
+    return lowest_bids.first if lowest_bids.length == 1
+    lowest_bids.sort_by(&:created_at).first
+  end
+
+  def winning_bidder_name
+    winning_bid.name
+  end
+
+  def winning_bid_amount
+    winning_bid.amount
+  end
+
+  def bid_is_winner?(bid)
+    return false if bid.nil? || available?
```

Comment:
I would break out methods to name some of these conditionals

`return false if ___`